src/components/tl/cuda/tl_cuda_coll.h

```diff
@@ -100,19 +113,39 @@ ucc_status_t ucc_tl_cuda_task_init(ucc_base_coll_args_t *coll_args,
         return status;
     }
 
-    task->seq_num = team->seq_num++;
-    task->coll_id = task->seq_num % max_concurrent;
+    /* active set */
+    if (UCC_COLL_ARGS_ACTIVE_SET(&coll_args->args)) {
+        ucc_assert(coll_args->args.coll_type == UCC_COLL_TYPE_BCAST);
+        task->subset.map    = ucc_active_set_to_ep_map(&coll_args->args);
+        task->subset.myrank = UCC_TL_TEAM_RANK(team);
+        // root
+        if (task->subset.myrank == coll_args->args.root) {
+            int peer               = ucc_ep_map_eval(task->subset.map, 1);
+            task->bcast_linear.key = ((uint64_t)coll_args->args.tag << 32 |
```

Comment:
also, I see that tag is only 16 bits:
https://github.com/openucx/ucc/blob/0c0fc21559835044ab107199e334f7157d6a0d3d/src/ucc/api/ucc_def.h#L135

So we could save space by shifting 48 bits instead of 32

Reply:
maybe it would be safer to also add some assert or static assert, that the sizes of the datatype do satisfy those underlying assumptions